### DISCUSSION TITLE
How to use the `_contains` JSONB operator?

### DISCUSSION THREAD

**Discussion by corysimmons** [hasura/graphql-engine#6375] [General] [answered]
I have a `nodes` table with an `access` JSONB column which lists the `owner` and `viewers` of that particular node.

![image](https://user-images.githubusercontent.com/224636/102714791-1ab16a00-429f-11eb-8e53-ccf30c3c8815.png)

I found the `_contains` operator in the Hasura API https://hasura.io/docs/1.0/graphql/core/api-reference/graphql-api/query.html#jsonb-operators

And looked at the Postgres equivalent `@>` docs: https://www.postgresql.org/docs/9.4/functions-json.html#FUNCTIONS-JSONB-OP-TABLE

The example the Postgres docs supplies says `'{"a":1, "b":2}'::jsonb @> '{"b":2}'::jsonb` which I assume would return `true` as `"b": 2` exists within the original JSON object.

What am I doing wrong in my query so `nodes` returns nothing?

Thank you!

**Comment by praveenweb** [comment, marked as answer]
@corysimmons - The `_contains` object can be written as 
```
{_contains: {owner: "bob"}}
```
In your example, the escaping of the object is the issue. If you click on `Analyze` to see the generated SQL, you would notice that Hasura already escapes the given expression for `_contains` operator and your usage makes it redundant and hence empty results.

**Reply by corysimmons** [reply]
Thank you so much Praveen! Love your tutorial vids btw. ;D

**Reply by corysimmons** [reply]
Hey @praveenweb

Is it possible to query on that `viewers` array? For instance, if I wanted to show `pat` all the `nodes` he had access to, could I write a query similar to:

```
nodes(where: {
  access: {
    viewers: {
      contains: "pat"
    }
  }
}) {
  id
}
```

?

That is... I'm trying to query like `someJson.someArray.includes('pat')`.
